aws-opentelemetry-distro/src/amazon/opentelemetry/distro/patches/_django_patches.py

```diff
@@ -136,3 +137,61 @@ def patched_uninstrument(self, **kwargs):
 
     except Exception as exc:  # pylint: disable=broad-exception-caught
         _logger.warning("Failed to apply Django code attributes patch: %s", exc)
+
+
+def _apply_django_rest_framework_patch() -> None:
+    """Django REST Framework patch for accurate code attributes
+
+    This patch specifically handles Django REST Framework ViewSets to provide
+    accurate code attributes that point to the actual ViewSet methods (list, create, etc.)
+    instead of the generic APIView.dispatch method.
+    """
+    try:
+        # Check if Django REST Framework is available
+        try:
+            import rest_framework  # pylint: disable=import-outside-toplevel,unused-import  # noqa: F401
+        except ImportError:
+            # DRF not installed, skip patching
+            _logger.debug("Django REST Framework not installed, skipping DRF code attributes patch")
+            return
+
+        from rest_framework.views import APIView  # pylint: disable=import-outside-toplevel
+        from rest_framework.viewsets import ViewSetMixin  # pylint: disable=import-outside-toplevel
+
+        from amazon.opentelemetry.distro.code_correlation import (  # pylint: disable=import-outside-toplevel
+            add_code_attributes_to_span,
+        )
+        from opentelemetry import trace  # pylint: disable=import-outside-toplevel
+
+        # Store original dispatch method
+        original_dispatch = APIView.dispatch
+
+        def patched_dispatch(self, request, *args, **kwargs):
+            """Patched dispatch method to add accurate code attributes for ViewSets."""
+            # Call original dispatch method first
+            response = original_dispatch(self, request, *args, **kwargs)
+
+            # Add code attributes if this is a ViewSet
+            try:
+                if isinstance(self, ViewSetMixin):
+                    span = trace.get_current_span()
+                    if span and span.is_recording():
+                        # Get the actual ViewSet method that will be executed
+                        action = getattr(self, "action", None)
+                        if action:
+                            # Get the actual method (list, create, retrieve, etc.)
+                            handler = getattr(self, action, None)
+                            if handler and callable(handler):
```

Comment:
Will following `add_code_attributes_to_span()` handle all kinds of `callable` input?

Reply:
The next line `add_code_attributes_to_span(span, handler)` is not a wrapper, it just add `handler` information into the `span`.